### PR TITLE
Update zf3

### PR DESCRIPTION
**Pull request by kokspflanze**
hi,
updated the system to zend-mvc 3 and zend-servicemanager 3

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,15 +19,15 @@
         }
     },
     "require": {
-        "php": ">=5.5",
-        "zendframework/zend-servicemanager": "~2.5",
-        "zendframework/zend-mvc": "~2.5",
-        "zendframework/zend-loader": "~2.5",
-        "zendframework/zend-config": "~2.5",
+        "php": ">=5.6",
```
Comment (samsonasik) [inline composer.json]: should support v2 ,so "php": "^5.5 | 7",

File: composer.json [hunk 2]
```diff
@@ -19,15 +19,15 @@
...
+        "zendframework/zend-servicemanager": "~3.0",
```
Comment (samsonasik) [inline composer.json]: "^2.5 | ^3.0"

File: composer.json [hunk 3]
```diff
@@ -19,15 +19,15 @@
...
+        "zendframework/zend-mvc": "~3.0",
```
Comment (samsonasik) [inline composer.json]: "^2.5 | ^3.0"

File: composer.json [hunk 4]
```diff
@@ -19,15 +19,15 @@
...
         "zendframework/zend-captcha": "~2.5",
-        "zendframework/zend-form": "~2.5",
-        "zendframework/zend-stdlib": "~2.5|~3.0",
-        "zendframework/zend-router": "~2.5",
-        "zendframework/zend-http": "~2.5"
+        "zendframework/zend-form": "~2.8",
+        "zendframework/zend-stdlib": "~3.0",
```
Comment (samsonasik) [inline composer.json]: "^2.5 | ^3.0"

File: composer.json [hunk 5]
```diff
@@ -19,15 +19,15 @@
...
+        "zendframework/zend-router": "~3.0",
+        "zendframework/zend-i18n": "~2.7",
+        "zendframework/zend-http": "~2.5",
+        "zendframework/zend-session": "~2.7"
```
Comment (samsonasik) [inline composer.json]: should support 2.5 for v2 compat: "^2.5 | ^3.0"

File: src/SanCaptcha/Controller/CaptchaControllerFactory.php [hunk 6]
```diff
@@ -2,21 +2,19 @@
...
     /**
-     * @param AbstractPluginManager|ServiceLocatorInterface $serviceLocator
-     *
+     * @param ContainerInterface $container
+     * @param string $requestedName
+     * @param array|null $options
      * @return CaptchaController
      */
-    public function createService(ServiceLocatorInterface $serviceLocator)
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
```
Comment (samsonasik) [inline src/SanCaptcha/Controller/CaptchaControllerFactory.php]: should support zend-servicemanager v2 too

File: src/SanCaptcha/Controller/TestcaptchaControllerFactory.php [hunk 7]
```diff
@@ -2,22 +2,19 @@
...
     /**
-     * @param AbstractPluginManager|ServiceLocatorInterface $serviceLocator
-     *
+     * @param ContainerInterface $container
+     * @param string $requestedName
+     * @param array|null $options
      * @return TestcaptchaController
      */
-    public function createService(ServiceLocatorInterface $serviceLocator)
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
```
Comment (samsonasik) [inline src/SanCaptcha/Controller/TestcaptchaControllerFactory.php]: should support zend-servicemanager v2 too

File: src/SanCaptcha/Service/CaptchaFactory.php [hunk 8]
```diff
@@ -2,21 +2,23 @@
...
 {
     /**
-     * @param ServiceLocatorInterface $sm
+     * @param ContainerInterface $container
+     * @param string $requestedName
+     * @param array|null $options
      * @return \Zend\Captcha\AdapterInterface
      */
-    public function createService(ServiceLocatorInterface $sm)
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
```
Comment (samsonasik) [inline src/SanCaptcha/Service/CaptchaFactory.php]: should support zend-servicemanager v2 too

File: src/SanCaptcha/Service/CaptchaFactory.php [hunk 9]
```diff
@@ -2,21 +2,23 @@
...
     {
-        $config = $sm->get('Config');
+        $config = $container->get('Config');
```
Comment (samsonasik) [inline src/SanCaptcha/Service/CaptchaFactory.php]: little 'config'

**Comment by kokspflanze** [comment]
okay, i will check what i can for v2 and v3 support in same version.

**Comment by kokspflanze** [comment]
updated the pr, so both versions should be now working.

but we have to move to v3 only if `Zend\ServiceManager\FactoryInterface` will be removed.

**Comment by samsonasik** [comment]
@kokspflanze merged, thank you ;), I've released tag 0.0.4, yes, at that time of `Zend\ServiceManager\FactoryInterface` removed,  we will only support v3

**Comment by samsonasik** [comment]
I'm releasing version 1.0.0 which drop zend-mvc v2 and zend-servicemanager v2 support, now with `Psr\Container\Container` type hint.

**Comment by kokspflanze** [comment]
@samsonasik okay=), maybe you can also update `ZF2 captcha Simple` in description

**Comment by samsonasik** [comment]
@kokspflanze thank you, I've updated the repo description

**Pull request merged**